patch/driver-hwmon-pmbus-dps200.patch

```diff
@@ -0,0 +1,169 @@
+ drivers/hwmon/pmbus/Makefile |    1 
+ drivers/hwmon/pmbus/dps200.c |  116 ++++++++++++++++++++++++++++++++++++++++++
+ 3 files changed, 127 insertions(+)
+ create mode 100644 drivers/hwmon/pmbus/dps200.c
+
+diff --git a/drivers/hwmon/pmbus/Kconfig b/drivers/hwmon/pmbus/Kconfig
+index 505ca2be0..fb4b0bbae 100644
+--- a/drivers/hwmon/pmbus/Kconfig
++++ b/drivers/hwmon/pmbus/Kconfig
+@@ -218,4 +218,14 @@ config SENSORS_DPS1900
+ 	  This driver can also be built as a module. If so, the module will
+ 	  be called dps1900.
+ 
++config SENSORS_DPS200
++	tristate "Delta DPS200"
++	default n
++	help
++	  If you say yes here you get hardware monitoring support for Delta
++	  DPS200.
++
++	  This driver can also be built as a module. If so, the module will
++	  be called dps200.
++
+ endif # PMBUS
+diff --git a/drivers/hwmon/pmbus/Makefile b/drivers/hwmon/pmbus/Makefile
+index c76820a4d..f072288f6 100644
+--- a/drivers/hwmon/pmbus/Makefile
++++ b/drivers/hwmon/pmbus/Makefile
+@@ -23,3 +23,4 @@ obj-$(CONFIG_SENSORS_UCD9000)	+= ucd9000.o
+ obj-$(CONFIG_SENSORS_UCD9200)	+= ucd9200.o
+ obj-$(CONFIG_SENSORS_ZL6100)	+= zl6100.o
+ obj-$(CONFIG_SENSORS_DPS1900)	+= dps1900.o
++obj-$(CONFIG_SENSORS_DPS200)	+= dps200.o
+diff --git a/drivers/hwmon/pmbus/dps200.c b/drivers/hwmon/pmbus/dps200.c
+new file mode 100644
+index 000000000..d11d455b3
+--- /dev/null
++++ b/drivers/hwmon/pmbus/dps200.c
+@@ -0,0 +1,116 @@
++// SPDX-License-Identifier: GPL-2.0-or-later
++/*
++ * Hardware monitoring driver for Delta DPS200
++ *
++ * (C) Copyright 2019, Celestica Inc.
++ * Author: Pradchaya Phucharoen <pphuchar@celestica.com>
++ */
++
++#include <linux/kernel.h>
++#include <linux/module.h>
++#include <linux/init.h>
++#include <linux/err.h>
++#include <linux/slab.h>
++#include <linux/mutex.h>
++#include <linux/i2c.h>
++#include <linux/pmbus.h>
++#include "pmbus.h"
++
++
++static int dps200_read_word_data(struct i2c_client *client, int page, int reg)
++{
++
++	if (reg >= PMBUS_VIRT_BASE ||
++	    reg == PMBUS_VOUT_OV_FAULT_LIMIT ||
++	    reg == PMBUS_VOUT_OV_WARN_LIMIT ||
++	    reg == PMBUS_VOUT_UV_WARN_LIMIT ||
++	    reg == PMBUS_VOUT_UV_FAULT_LIMIT ||
++	    reg == PMBUS_IOUT_OC_LV_FAULT_LIMIT ||
++	    reg == PMBUS_IOUT_UC_FAULT_LIMIT ||
++	    reg == PMBUS_UT_WARN_LIMIT ||
++	    reg == PMBUS_UT_FAULT_LIMIT ||
++	    reg == PMBUS_VIN_OV_FAULT_LIMIT ||
++	    reg == PMBUS_VIN_OV_WARN_LIMIT ||
++	    reg == PMBUS_IIN_OC_FAULT_LIMIT ||
++	    reg == PMBUS_POUT_MAX)
++		return -ENXIO;
++
++	/*
++	 * NOTE: The following feild have constant driver value.
++	 * If the register are PMBUS_IOUT_OC_WARN_LIMIT or
++	 * PMBUS_IOUT_OC_FAULT_LIMIT convert the constant value from linear
++	 * to direct format.
++	 */
++	if (reg == PMBUS_IOUT_OC_WARN_LIMIT)
++		return 0xb4;
++	else if (reg == PMBUS_IOUT_OC_FAULT_LIMIT)
++		return 0xc8;
++	else
++		return pmbus_read_word_data(client, page, reg);
++}
++
++/*
++ * Form DPS-200 datasheet the supported sensors format defined as:
```

Comment:
s/Form/From/

Reply:
> Also, judging from `$ diff -u drivers/hwmon/pmbus/dps1900.c drivers/hwmon/pmbus/dps200.c` both are pretty similar.
> 
> Does DPS1900 have a bug?
> 
> ```
> 	.read_word_data = dps1900_read_word_data,
> ```
> 
> Should that be `&dps1900_read_word_data`?

Both code style is the same result. 
To make it the same style as dps1900. I have remove `&`.



> s/Form/From/

I have updated typo in the file.